tripal_chado/tests/src/Functional/Services/ChadoTripalPublishTest.php

```diff
@@ -0,0 +1,590 @@
+
+
+    $this->checkFieldItem('organism', 'organism_genus', 1,
+        ['record_id' => $organism_id2],
+        ['bundle' => 'organism', 'entity_id' => 2, 'value' => NULL]);
+
+    $this->checkFieldItem('organism', 'organism_species', 1,
+        ['record_id' => $organism_id2],
+        ['bundle' => 'organism', 'entity_id' => 2, 'value' => NULL]);
+
+    $this->checkFieldItem('organism', 'organism_abbreviation', 1,
+        ['record_id' => $organism_id2],
+        ['bundle' => 'organism', 'entity_id' => 2, 'value' => NULL]);
+
+    $this->checkFieldItem('organism', 'organism_infraspecific_name', 0,
+        ['record_id' => $organism_id2],
+        ['bundle' => 'organism', 'entity_id' => 2, 'value' => NULL]);
+
+    $this->checkFieldItem('organism', 'organism_infraspecific_type', 1,
+        ['record_id' => $organism_id2],
+        ['bundle' => 'organism', 'entity_id' => 2, 'type_id' => NULL]);
+
+    $this->checkFieldItem('organism', 'organism_comment', 1,
+        ['record_id' => $organism_id2],
+        ['bundle' => 'organism', 'entity_id' => 2, 'value' => NULL]);
+
+    //
+    // Test publishing properties.
+    //
+    $comment_type_id = $schema_db->getTerm('comment')->getInternalId();
+    $note_type_id = $local_db->getTerm('Note')->getInternalId();
+    $this->attachOrganismPropertyFields();
+    $this->addProperty($chado, 'organism', [
+      'organism_id' => $organism_id,
+      'type_id' => $note_type_id,
+      'value' => 'Note 1',
+      'rank' => 1,
+    ]);
+    $this->addProperty($chado, 'organism', [
+      'organism_id' => $organism_id,
+      'type_id' => $note_type_id,
+      'value' => 'Note 0',
+      'rank' => 0,
+    ]);
+    $this->addProperty($chado, 'organism', [
+      'organism_id' => $organism_id,
+      'type_id' => $note_type_id,
+      'value' => 'Note 2',
+      'rank' => 2,
+    ]);
+
+    $this->addProperty($chado, 'organism', [
+      'organism_id' => $organism_id,
+      'type_id' => $comment_type_id,
+      'value' => 'Comment 0',
+      'rank' => 0,
+    ]);
+    $this->addProperty($chado, 'organism', [
+      'organism_id' => $organism_id,
+      'type_id' => $comment_type_id,
+      'value' => 'Comment 1',
+      'rank' => 1,
+    ]);
+
+
+    // Now publish the organism content type again.
+    $publish->init('organism', 'chado_storage');
+    $entities = $publish->publish();
+
+    // Because we added properties for the first organism we should set it's
+    // entity in those returned, but not the gorilla organism.
+    $this->assertTrue(array_values($entities)[0] == '<em>Oryza species</em> subspecies <em>Japonica</em>',
+      'The Oryza species subspecies Japonica organism should appear in the published list because it has new properties.');
+    $this->assertTrue(count(array_values($entities)) == 1,
+      'There should only be one published entity for a single organism with new properties.');
```

Comment:
Using assertCount will show the actual count obtained if it fails whereas using assertTrue and doing the comparison yourself does not provide that information.
```suggestion
    $this->assertCount(1, array_values($entities),
      'There should only be one published entity for a single organism with new properties.');
```